tests/vector/test_spaces.py

```diff
@@ -129,3 +133,78 @@ def test_iterate_custom_space(space, batch_space):
     for i, item in enumerate(iterator):
         assert item in space
     assert i == 3
+
+
+seeded_spaces = [
+    CustomSpace(seed=123),
+    Box(0, 10, (), seed=123),
+    Tuple([Box(0, 5, (), seed=123), Box(0, 3, (), seed=123)], seed=123),
+    Dict(
+        {"space-1": Box(0, 5, (), seed=123), "space-2": Box(0, 10, (), seed=123)},
+        seed=123,
+    ),
+    Discrete(5, seed=123),
+    MultiDiscrete([5, 3], seed=123),
+]
+
+
+@pytest.mark.parametrize(
+    "space", seeded_spaces, ids=[space.__class__.__name__ for space in seeded_spaces]
+)
+def test_batch_space_seed(space):
+    batched_space = batch_space(space)  # n=1
+    assert space.np_random == batched_space.np_random
+
+
+@pytest.mark.parametrize(
+    "space", spaces, ids=[space.__class__.__name__ for space in spaces]
+)
+@pytest.mark.parametrize("n", [4, 5], ids=[f"n={n}" for n in [3, 5]])
+@pytest.mark.parametrize(
+    "base_seed", [123, 456], ids=[f"seed={base_seed}" for base_seed in [123, 456]]
+)
+def test_rng_different_at_each_index(space, n, base_seed):
+    space.seed(base_seed)
+
+    batched_space = batch_space(space, n)
+    assert space.np_random == batched_space.np_random
+
+    batched_sample = batched_space.sample()
+    sample = list(iterate(batched_space, batched_sample))
+    assert not all(np.all(element == sample[0]) for element in sample), sample
+
+
+@pytest.mark.parametrize(
+    "space", spaces, ids=[space.__class__.__name__ for space in spaces]
+)
+@pytest.mark.parametrize("n", [1, 2, 5], ids=[f"n={n}" for n in [1, 2, 5]])
+@pytest.mark.parametrize(
+    "base_seed", [123, 456], ids=[f"seed={base_seed}" for base_seed in [123, 456]]
+)
+def test_deterministic(space: Space, n: int, base_seed: int):
+    space_a = space
+    space_a.seed(base_seed)
+    space_b = copy.deepcopy(space_a)
+    assert space_a.np_random == space_b.np_random
+    assert space_a.np_random is not space_b.np_random
+
+    space_a_batched = batch_space(space_a, n)
+    space_b_batched = batch_space(space_b, n)
+    assert space_a_batched == space_b_batched
+    assert space_a_batched.np_random == space_b_batched.np_random
+    assert space_a_batched.np_random is not space_b_batched.np_random
+    assert space_a.np_random is not space_a_batched.np_random
+
+    # Check that batched space a and b random number generator are not effected by the original space
+    space_a.sample()
+    space_a_batched_sample = space_a_batched.sample()
+    space_b_batched_sample = space_b_batched.sample()
+    for a_sample, b_sample in zip(
+        iterate(space_a_batched, space_a_batched_sample),
+        iterate(space_b_batched, space_b_batched_sample),
+    ):
+        if isinstance(a_sample, tuple):
+            for a_subsample, b_subsample in zip(a_sample, b_sample):
```

Comment:
Really subtle, but here you should probably assert that they have the same length, or use `itertools.zip_longest` instea of `zip`, otherwise the test could hypothetically silently pass when the samples are different.
```suggestion
            for a_subsample, b_subsample in itertools.zip_longest(a_sample, b_sample):
```
(edit: oh and you'd need to import itertools at the top of the file too)

Reply:
We expect them to be the same size, yes? I'd just put an assert on the length in that case

Reply:
I'd expect that `iterate` might return an iterator rather than a list or tuple, so calling `len` would exhaust it. That's why I'm thinking zip_longest might be safer, since IIRC the for loop over an exhausted iterator will just do nothing? or raise an exception? not sure.

Reply:
This can be marked as resolved.